### PR TITLE
Be 05 add comments

### PR DESCRIPTION
**Pull request by DanielOldak**

### CODE REVIEW

File: app/controllers/api/v1/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+module Api
+  module V1
+    class CommentsController < ApplicationController
+      def create
+        form = Api::V1::CommentsCreateForm.new(comment_params.merge(recipe: recipe))
+        if form.save
+          render json: form.comment
+        else
+          render json: { errors: form.errors.full_messages }, status: :unprocessable_entity
```
Comment (PatrykTyszka) [inline app/controllers/api/v1/comments_controller.rb:9]: did you ask if frontend wants to get errors in this format?

File: app/forms/api/v1/comments_create_form.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module Api
+  module V1
+    class CommentsCreateForm
+      include ActiveModel::Model
```
Comment (PatrykTyszka) [inline app/forms/api/v1/comments_create_form.rb:4]: 1. First of all nice to see that someone knows what FormObject is and how to use it
2. Good job with namespace 

BUT 

it's overgrown with content... look at the create_recipe_comment. You could add this line inside the controller and move these validations to the model. It's even required to have these validations inside the model coz someone can create invalid data via rails console (you should prevent that as well). I would recommend removing this class ( even though I love it ) and move this logic to the controller coz at this moment it's to much :)

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  # validations included in Api::V1::CommentCreateForm
```
Comment (PatrykTyszka) [inline app/models/comment.rb:2]: remove this comment and add validations here. Do not forget to add knowde's comments e.g 
```
  class SomeModel
    # == Constants ============================================================

    # == Extensions ===========================================================

    # == Relationships ========================================================

    # == Validations ==========================================================

    # == Scopes ===============================================================

    # == Callbacks ============================================================

    # == Nested Attributes ====================================================

    # == Class Methods ========================================================

    # == Instance Methods =====================================================
  end
```

File: app/serializers/recipe_serializer.rb [hunk 4]
```diff
@@ -1,3 +1,10 @@
 class RecipeSerializer < ActiveModel::Serializer
   attributes :id, :name, :content
+
+  has_many :comments
+  has_many :comments, serializer: Api::V1::CommentSerializer
```
Comment (PatrykTyszka) [inline app/serializers/recipe_serializer.rb:5]: not sure if you need these two lines if you have `def comment` please check

File: spec/controllers/api/v1/comments_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+describe Api::V1::CommentsController do
+  describe '[POST] #create' do
+    let(:recipe) { create(:recipe) }
+    let(:comment) { build(:comment) }
+    let(:comment_params) { { author: comment.author, body: comment.body } }
```
Comment (PatrykTyszka) [inline spec/controllers/api/v1/comments_controller_spec.rb:7]: Do not use a factory for creating params of comments. Do it explicitly using let's

File: spec/forms/api/v1/comments_create_form_spec.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+describe Api::V1::CommentsCreateForm do
```
Comment (PatrykTyszka) [inline spec/forms/api/v1/comments_create_form_spec.rb:3]: ehh :P you know what... wait for @h4b00 review :d maybe he will say that we should keep this Form Object :D

File: spec/serializers/api/v1/comment_serializer_spec.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+describe Api::V1::CommentSerializer do
+  subject(:serialized_comment) { described_class.new(comment) }
+
+  let(:comment) do
+    recipe = create(:recipe)
+    recipe.comments << build(:comment)
+    recipe.save
+    recipe.comments.first
+  end
```
Comment (PatrykTyszka) [inline spec/serializers/api/v1/comment_serializer_spec.rb:11]: use factories for that -.-

**Review by PatrykTyszka — CHANGES_REQUESTED** [review]

**Review by h4b00 — CHANGES_REQUESTED** [review]

**Review by PatrykTyszka — APPROVED** [review]

**Review by PatrykTyszka — APPROVED** [review]

**Review by h4b00 — APPROVED** [review]
GJ

**Pull request merged**